Extensions/quick_tags.js

```diff
@@ -397,21 +399,27 @@ XKit.extensions.quick_tags = new Object({
 
 	},
 
-	do_posts: function() {
-
-		// get posts:
-		var posts = XKit.interface.get_posts("xkit-quick-tags-done", true, true);
-
-		$(posts).each(function() {
+	do_posts: async function() {
+		if (XKit.extensions.quick_tags.processing === true) {
+			return;
+		}
 
-			$(this).addClass("xkit-quick-tags-done");
+		if (XKit.interface.where().inbox) {
+			return;
+		}
 
-			if ($(this).hasClass("is_note") && XKit.interface.where().inbox === true) { return; }
+		XKit.extensions.quick_tags.processing = true;
+		var $posts = await XKit.interface.react.get_posts("xkit-quick-tags-done", true);
 
-			XKit.interface.add_control_button(this, "xkit-quick-tags", "");
+		$posts
+			.addClass("xkit-quick-tags-done")
+			.each(function() {
+				const $post = $(this);
 
-		});
+				XKit.interface.react.add_control_button($post, "xkit-quick-tags", "");
```

Comment:
the only reason to save a jQuery object as a variable is to avoid calling the constructor multiple times; since it's only used here once, this function can just be `XKit.interface.react.add_control_button($(this), "xkit-quick-tags", "");`

(cant code-suggest thanks to the unchanged newline, great job github)